BUILD

```diff
@@ -4325,23 +4325,49 @@ grpc_cc_library(
     ],
 )
 
+grpc_cc_library(
+    name = "tsi_ssl_session_cache",
+    srcs = [
+        "src/core/tsi/ssl/session_cache/ssl_session_boringssl.cc",
+        "src/core/tsi/ssl/session_cache/ssl_session_cache.cc",
+        "src/core/tsi/ssl/session_cache/ssl_session_openssl.cc",
+    ],
+    hdrs = [
+        "src/core/tsi/ssl/session_cache/ssl_session.h",
+        "src/core/tsi/ssl/session_cache/ssl_session_cache.h",
+    ],
+    external_deps = [
+        "absl/strings",
+        "libssl",
+        "libcrypto",
+    ],
+    language = "c++",
+    visibility = ["@grpc:public"],
+    deps = [
```

Comment:
Overall this is looking great! Did you get a chance to clean up the unused dependencies here as we discussed last time?